### DISCUSSION TITLE
Custom layouts don't load correctly

### DISCUSSION THREAD

**Discussion by bkerz** [zellij-org/zellij#2226] [Q&A] [answered]
Hi everyone, I would appreciate if you guys can help me to create custom layouts. I'm using Alacritty, Nushell and Zellij (0.34.4), I configured everything correctly (i think) but when I want to load a layout, it loads inside the current pane and it doesn't show the layout I defined either. I tried what the documentation says a lot of time (which honestly it's not that much) but It still doesn't work. Here is what `zellij setup --check` says (maybe this info is useful):

![image](https://user-images.githubusercontent.com/48168047/223091073-2b370d4a-af6c-434d-a2cd-eef4c049e5c3.png)

Under `~/config/zellij/layouts` I have two layouts:

![image](https://user-images.githubusercontent.com/48168047/223091446-27f97028-71f8-43f5-9a5f-84dd18cf021a.png)

This is what I have in next.kdl:

![image](https://user-images.githubusercontent.com/48168047/223091603-ad57670a-1924-4eee-9dac-c8fda522b1c7.png)

and this happens when I run `zellij --layout next`

![sketchpad pro_sp26ecwt4jo079qt8qe2w_p1](https://user-images.githubusercontent.com/48168047/223092614-8d2295c3-ab4f-4f81-a198-5ec59b268b98.png)

And something similar happens with the other layout. Any idea of what's happening? Thanks in advance.

**Comment by bkerz** [comment]
Another thing is that when I try to open a new tab with one of the layouts it doesn't seem able to find them:

![image](https://user-images.githubusercontent.com/48168047/223097612-9d6a802e-51a5-4b8a-a3e1-c381cf03324c.png)

**Comment by imsnif** [comment, marked as answer]
Hey, loading a layout with `--layout` is something you need to do when outside of Zellij. eg. "Please open Zellij with the following layout". You can load a single-tab layout from within Zellij with the `new-tab` action, as you mentioned.

The new-tab action for now indeed doesn't find layouts in the default layout folder - this is a bug that's fixed in main and will be released in the next version (hopefully tomorrow if all goes well!)

I hope this answers your concerns.

**Reply by bkerz** [reply]
Yep, thanks for the time and the answer, the only detail is that I'm experiencing the same issue as #2217, after reading for a while and trial and error I finally managed to do what I wanted to do at the beginning with this layout:

```
layout {
     tab name="Next" {
        pane size=1 borderless=true {
            plugin location="zellij:tab-bar"
        }
        pane
        pane size=2 borderless=true {
            plugin location="zellij:status-bar"
        }
     }
     tab name="Nvim"{
        pane size=1 borderless=true {
            plugin location="zellij:tab-bar"
        }
        pane
        pane size=2 borderless=true {
            plugin location="zellij:status-bar"
        }
     }
     tab name="Git"{
        pane size=1 borderless=true {
            plugin location="zellij:tab-bar"
        }
        pane
        pane size=2 borderless=true {
            plugin location="zellij:status-bar"
        }
     }
}
```
It's be good to avoid writing the default plugins, but I guess you guys will work on this at some point. Well I think that's all, thanks again for the help.
